### ISSUE TITLE
Investigate Provenance build steps on M1 w Xcode 12.5

### ISSUE THREAD

**Issue by mrjschulte** [Provenance-Emu/Provenance#1527]
Currently the install process for mainline Develop branch of Provenance and the Carthage pre-build stage don't work on M1 with Xcode 12.5+. Lets try and Address this with the needed Carthage dependency updates and the various commands needed to get a successful build.

This will be a investigation/WIP Issue and the resulting PR's will be linked into this Issue until we lock in a solid and proven workflow to build out Pv on M1 systems.

The WIP Branch to try and address the items in this issue lives here:
https://github.com/mrjschulte/Provenance/tree/xCode_12_5_update

**Comment by mrjschulte** [comment]
Bump up Carthage sub projects:
RxSwift to 5.1.2
realm-cocoa 5.5.2

**Comment by mrjschulte** [comment]
Set RxRealm back to the official GitHub release and bump it up to RxSwiftCommunity/RxRealm" "v5.0.2"

**Comment by mrjschulte** [comment]
"Clean" Carthage build of tvOS was successful
./carthage.sh bootstrap --use-xcframeworks --platform tvOS  188.73s user 76.62s system 60% cpu 7:21.59 total

**Comment by mrjschulte** [comment]
Looks like using the --use-xcframeworks also causes Carthage to NOT place the frameworks in the correct iOS/tvOS directory in the Carthage/Build dir. Not moving the respective built frameworks into that relative directory in Builds will cause the Xcode build to fail. 

Workaround: Make a Carthage/Build/iOS and Carthage/Build/tvOS directory accordingly before running The Carthage bootstrap commands.

**Comment by mrjschulte** [comment]
Looks like Xcode 12.5 requires the use of xcframeworks so we'll need to see about how to add support for that.

**Comment by mrjschulte** [comment]
Specifying --platform iOS,tvOS will build the needed .xcframeworks in one go and in the same Build dir. The individual iOS / tvOS dirs we had previously will not be used.

**Comment by mrjschulte** [comment]
Ok, we now have a *very* experimental branch that successfully builds on M1 with Xcode 12.5 using XCframeworks and the latest Carthage.

**Comment by mrjschulte** [comment]
One thing I've noticed is that I can not successfully deploy and launch on a prior tvOS version (before 14.6) after building with Xcode 12.5 on M1. You can launch the app whilst tethered to Xcode just fine for development, but as soon as you disconnect from Xcode I can no longer launch Pv directly on the tvOS that ran the app fine a few seconds prior.

**Comment by mrjschulte** [comment]
So, IF you wanted to help test and give Pv a first ALPHA like spin whilst building on Xcode 12.5 on the M1, do the following and let me know how it goes:

Install homebrew, if you haven't already.
Then install/update Carthage:
brew upgrade carthage

Then create a new dir for this build, cd into it and clone my repo with the refactor:
git clone -b xCode_12_5_update -j 24 --recurse https://github.com/mrjschulte/Provenance.git

Then in the root of the Provenance folder that gets created run:
arch -x86_64 ./carthage.sh bootstrap --platform iOS,tvOS --no-use-binaries --use-xcframeworks

And get a cup of coffee, or two. Took about 26minutes on my 8Core M1 Mini.

When Carthage is done load up the Provenance.xcworkspace and change the various entitlements as needed to build Pv to AppleTV or iOS w Xcode 12.5 on a M1 Mini.

**Comment by loicpirez** [comment]
Hello @mrjschulte  !
I was able to build too with your branch on Xcode 12.5.1 with an Intel based Mac.
This build run on my iOS devices, and if you need to test specific things I can help,
Thanks again !

**Comment by mrjschulte** [comment]
Glad it worked out for you. Feel free to use this branch until we merge  main line develop. After which this branch will go away.

**Comment by loicpirez** [comment]
> Glad it worked out for you. Feel free to use this branch until we merge main line develop. After which this branch will go away.

No problem! Seems pretty stable so far.

**Comment by natalia-osa** [comment]
Maybe it'd be good to migrate to Swift Packages? It may be a bit longer for CI to build, but probably a lot less troublesome for users to install.

**Comment by JoeMatt** [comment]
> Maybe it'd be good to migrate to Swift Packages? It may be a bit longer for CI to build, but probably a lot less troublesome for users to install.

We're testing that. We had to update some dependencies first to even begin to support it since older versions of packages didn't support swift pm.

Me and @mrjschulte have discussed whether we should do this first or do a release then migrate.
I was able to update all the dependencies to a current version and I think that's a good stopping point to wrap up other fixes, maybe do a beta release, then evaluate the work needed to change package systems.

If you want to give it a shot, would love a PR, even if it's a proof of concept 🙂
